pkg/api/v1alpha1/cluster_webhook_test.go

```diff
@@ -1442,23 +1149,242 @@ func TestClusterUpdateControlPlaneTaintsAndLabelsSuccess(t *testing.T) {
+}
+
+func TestClusterValidateCreateManagedUnpause(t *testing.T) {
+	features.ClearCache()
+	cluster := createCluster()
+	g := NewWithT(t)
+	cluster.SetManagedBy("mgmt2")
+	validateCreateError := cluster.ValidateCreate()
+	g.Expect(validateCreateError).NotTo(Succeed())
+	g.Expect(validateCreateError.Error()).To(ContainSubstring("cluster on existing cluster is not supported"))
+}
+
+func TestClusterValidateCreateSelfManagedNotPaused(t *testing.T) {
+	features.ClearCache()
+	t.Setenv(features.FullLifecycleAPIEnvVar, "true")
+	cluster := createCluster()
+	cluster.SetSelfManaged()
+
+	g := NewWithT(t)
+	g.Expect(cluster.ValidateCreate().Error()).To(ContainSubstring("not supported for self managed clusters"))
+}
+
+func TestClusterValidateCreateInvalidCluster(t *testing.T) {
+	tests := []struct {
+		name               string
+		featureGateEnabled bool
+		cluster            *v1alpha1.Cluster
+	}{
+		{
+			name: "Paused self-managed cluster, feature gate off",
+			cluster: newCluster(func(c *v1alpha1.Cluster) {
+				c.SetSelfManaged()
+				c.PauseReconcile()
+			}),
+			featureGateEnabled: false,
+		},
+		{
+			name: "Unpaused self-managed cluster, feature gate off",
+			cluster: newCluster(func(c *v1alpha1.Cluster) {
+				c.SetSelfManaged()
+			}),
+			featureGateEnabled: false,
+		},
+		{
+			name: "Paused workload cluster, feature gate off",
+			cluster: newCluster(func(c *v1alpha1.Cluster) {
+				c.SetManagedBy("my-management-cluster")
+				c.PauseReconcile()
+			}),
+			featureGateEnabled: false,
+		},
+		{
+			name: "Unpaused workload cluster, feature gate off",
+			cluster: newCluster(func(c *v1alpha1.Cluster) {
+				c.SetManagedBy("my-management-cluster")
+			}),
+			featureGateEnabled: false,
+		},
+		{
+			name: "Paused self-managed cluster, feature gate on",
+			cluster: newCluster(func(c *v1alpha1.Cluster) {
+				c.SetSelfManaged()
+				c.PauseReconcile()
+			}),
+			featureGateEnabled: true,
+		},
+		{
+			name: "Unpaused self-managed cluster, feature gate on",
+			cluster: newCluster(func(c *v1alpha1.Cluster) {
+				c.SetSelfManaged()
+			}),
+			featureGateEnabled: true,
+		},
+		{
+			name: "Paused workload cluster, feature gate on",
+			cluster: newCluster(func(c *v1alpha1.Cluster) {
+				c.SetManagedBy("my-management-cluster")
+				c.PauseReconcile()
+			}),
+			featureGateEnabled: true,
+		},
+		{
+			name: "Unpaused workload cluster, feature gate on",
+			cluster: newCluster(func(c *v1alpha1.Cluster) {
+				c.SetManagedBy("my-management-cluster")
+			}),
+			featureGateEnabled: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			features.ClearCache()
+			if tt.featureGateEnabled {
+				t.Setenv(features.FullLifecycleAPIEnvVar, "true")
+			}
+
+			// Invalid control plane configuration
+			tt.cluster.Spec.ControlPlaneConfiguration = v1alpha1.ControlPlaneConfiguration{}
+
+			g := NewWithT(t)
+			g.Expect(tt.cluster.ValidateCreate()).To(MatchError(ContainSubstring("invalid cluster")))
```

Comment:
nit: why don't we look for the specific error we get with an invalid control plane? It seems a bit more reliable